varcode/vcf.py

```diff
@@ -58,42 +171,307 @@ def load_vcf(
+
+    Returns
+    ---------
+    If chunk_size is None (the default), a dataframe with the contents of the
+    VCF file. Otherwise, an iterable of dataframes, each with chunk_size rows.
+
     """
+    vcf_field_types = collections.OrderedDict()
+    vcf_field_types['CHROM'] = str
+    vcf_field_types['POS'] = int
+    vcf_field_types['ID'] = str
+    vcf_field_types['REF'] = str
+    vcf_field_types['ALT'] = str
+    vcf_field_types['QUAL'] = str
+    vcf_field_types['FILTER'] = str
+    if include_info:
+        vcf_field_types['INFO'] = str
 
-    typechecks.require_string(path, "Path to VCF")
+    parsed_path = parse_url_or_path(path)
+    if not parsed_path.scheme or parsed_path.scheme.lower() == "file":
+        path = parsed_path.path
+    else:
+        raise NotImplementedError("Only local files are supported.")
 
-    vcf_reader = vcf.Reader(filename=path)
+    reader = pandas.read_table(
+        path,
+        compression='infer',
+        comment="#",
+        chunksize=chunk_size,
+        dtype=vcf_field_types,
+        names=list(vcf_field_types),
+        usecols=range(len(vcf_field_types)))
+    return reader
+
+class PyVCFReaderFromPathOrURL(object):
+    '''
+    Thin wrapper over a PyVCF Reader object that supports loading over URLs,
+    and a close() function (pyvcf somehow doesn't have a close() funciton).
+
+    Attributes
+    ----------
+    path : string or None
+        path that was loaded, if available.
+
+    vcf_reader : pyvcf Reader instance
+    '''
+    def __init__(self, path):
+        """
+        Construct a new wrapper.
+
+        Parameters
+        ----------
+        path : string or pyvcf Reader instance
+            Path or URL to load, or Reader instance.
+        """
+        self.path = None  # string path, if available.
+        self.vcf_reader = None  # vcf_reader. Will always be set.
+        self._to_close = None  # object to call close() on when we're done.
+
+        if isinstance(path, vcf.Reader):
+            self.vcf_reader = path
+        else:
+            typechecks.require_string(path, "Path or URL to VCF")
+            self.path = path
+            parsed_path = parse_url_or_path(path)
+            if not parsed_path.scheme or parsed_path.scheme.lower() == 'file':
+                self.vcf_reader = vcf.Reader(filename=parsed_path.path)
+            elif parsed_path.scheme.lower() in ("http", "https", "ftp"):
+                self._to_close = response = requests.get(path, stream=True)
+                response.raise_for_status()  # raise error on 404, etc.
+                if path.endswith(".gz"):
+                    lines = stream_gzip_decompress_lines(
+                        response.iter_content())
+                else:
+                    lines = response.iter_lines(decode_unicode=True)
+                self.vcf_reader = vcf.Reader(fsock=lines, compressed=False)
+            else:
+                raise ValueError("Unsupported scheme: %s" % parsed_path.scheme)
```

Comment:
Oh, I've always called those protocols: https://en.wikipedia.org/wiki/URI_scheme (which wikipedia tells me is incorrect)